frontend/src/App.tsx

```diff
@@ -1,43 +1,86 @@
 
+    const components = [{
+      title: 'Message Overview',
+      page: <MessageOverview eventBus={this.eventBus} loginManager={this.loginManager} />,
+      selected: true,
+      id: 0,
+      icon: ImageProvider.getImage("home-icon"),
+      path: "/overview"
+    }, {
+      id: 1,
+      title: 'Analytics',
+      page: <AnalyticsPage eventBus={this.eventBus} />,
+      icon: ImageProvider.getImage("analytics"),
+      path: "/analytics"
+    }, {
+      id: 2,
+      title: 'Settings',
+      page: <SettingsPage eventBus={this.eventBus} loginManager={this.loginManager} />,
+      icon: ImageProvider.getImage("settings-icon"),
+      path: "/settings"
+    }, {
+      id: 3,
+      title: 'About',
+      page: <AboutPage />,
+      icon: ImageProvider.getImage("info-icon"),
+      path: "/about"
+    }];
+
+    const routes = components.forEach(c => {
+      return <Route path={c.path} render={(props) => this.loginManager.isLoggedIn() ? this.selectComponent(components, c.id) : <Redirect to={"/login?redirect=" + encodeURIComponent(c.path)} />} />;
+    });
     // /login -> go to Login Page
     // everything else -> Go to Supersocial application
     return (
       <BrowserRouter>
-        <Switch>  
-          <Route path="/register" render={(props) => <Registration username="" password="" email="" eventBus={eventBus} loginManager={loginManager} />} />        
-          <Route path="/login" render={(props) => <Login username="" password="" eventBus={eventBus} loginManager={loginManager} />} />
-          {this.state.loggedIn ? <Route path="/" render={(props) => <Supersocial eventBus={eventBus} loginManager={loginManager} />} /> : <Redirect to="/login" />} />}
+        <Switch>
+          <Route path="/register" render={(props) => <Registration eventBus={this.eventBus} loginManager={this.loginManager} />} />
+          <Route path="/login" render={(props) => <Login eventBus={this.eventBus} loginManager={this.loginManager} params={props.location.search} />} />
+          <Route path="/overview" render={(props) => this.loginManager.isLoggedIn() ? this.selectComponent(components, 0) : <Redirect to={"/login?redirect=" + encodeURIComponent("/overview")} />} />
+          <Route path="/analytics" render={(props) => this.loginManager.isLoggedIn() ? this.selectComponent(components, 1) : <Redirect to={"/login?redirect=" + encodeURIComponent("/analytics")} />} />
+          <Route path="/settings" render={(props) => this.loginManager.isLoggedIn() ? this.selectComponent(components, 2) : <Redirect to={"/login?redirect=" + encodeURIComponent("/settings")} />} />
+          <Route path="/about" render={(props) => this.loginManager.isLoggedIn() ? this.selectComponent(components, 3) : <Redirect to={"/login?redirect=" + encodeURIComponent("/about")} />} />
+          <Route path="/privacy" render={(props) => <DataPrivacyPage />} />
+          <Route path="/" render={(props) => <LandingPage />} />
```

Comment:
Then, this could be simplified to
```suggestion
          <Route path="/register" render={(props) => <Registration eventBus={this.eventBus} loginManager={this.loginManager} />} />
          <Route path="/login" render={(props) => <Login eventBus={this.eventBus} loginManager={this.loginManager} params={props.location.search} />} />
          {components.map((component, idx) => <Route path={component.path} render={(props) => !component.checkLogin || this.loginManager.isLoggedIn() ? this.selectComponent(components, idx) : <Redirect to={"/login?redirect=" + encodeURIComponent(component.path)} />} />
          )}
          <Route path="/privacy" render={(props) => <DataPrivacyPage />} />
          <Route path="/" render={(props) => <LandingPage />} />
```

Reply:
What is component.checkLogin supposed to be?

Reply:
So that you can decide whether you need to check `this.loginManager.isLoggedIn()` or not.